test/client_ui_test.py

```diff
@@ -84,6 +85,111 @@ def test_draw():
     assert f.last == line.ljust(79) + "\n"
 
 
+def test_json_format():
+    fake_time = FakeTime()
+    f = FakeFile()
+
+    # Size is unknown at this point.
+    pb = client.ProgressBar(
+        phase="setting up", output=f, format="json", now=fake_time)
+    line_data = {
+        "transferred": 0,
+        "size": None,
+        "elapsed": 0.0,
+        "description": "setting up",
+    }
+    line = f'{json.dumps(line_data)}'
+    assert f.last == line + "\n"
+
+    # Size was updated, but no bytes were transferred yet.
+    fake_time.now += 0.1
+    pb.size = 3 * GiB
+    line_data = {
+        "transferred": 0,
+        "size": 3 * GiB,
+        "elapsed": fake_time.now,
+        "description": "setting up",
+    }
+    line = f'{json.dumps(line_data)}'
+    assert f.last == line + "\n"
```

Comment:
First 2 tests looks good:
- size unknown
- all values known

What is the purpose of the rest of the tests?

Reply:
I'm not sure the current test is helpful, but if you are not sure it is net lets keep it as is.